tests/nuodb_cursor_test.py

```diff
@@ -125,6 +127,29 @@ def test_executemany_somefail(self):
 
         cursor.execute("DROP TABLE executemany_table")
 
+    def test_result_set_gets_closed(self):
+        current_version = getenv('NUODB_VERSION', None) # skipif <2.1
+        if current_version is not None and not LooseVersion(current_version) < LooseVersion("2.1"):
+        # Server will throw error after 1000 open result sets
+            con = self._connect()
+            for j in [False, True]:
```

Comment:
The first loop runs 2015 queries with the cursor being explicitly closed at the end. It is expected that nothing will happen (ie no error thrown) because the cursor is being closed. This mimics the original code snippet provided in the PR where a cursor.close was being called but ResultSets still remained open.

The second loop runs 2015 queries and expects a specific error to be thrown. In this case we're expecting an OperationalError to be thrown after the 1000th iteration. Notice how the second case doesn't call cursor.close(). 

When I performed testing under 2.0.4 within Travis and under my Local Machine I was able to continue executing queries with a ResultSet without cleaning up via a cursor.close. I'm not too familiar with how we handled ResultSets beforehand - but yes from what I noticed I could execute more than 1000 queries without having to call close on the cursor.